### ISSUE TITLE
set up git clone and import the repos as modules

### ISSUE THREAD

**Issue by wgyt** [wgytcraft/wgytcraft-hosting#2]
# set up modules
## clone the modules from git uri
I would use [NodeGit](https://NodeGit.org) for this but we could use something else
We would clone the module in to the modules folder:
example:
```json
{
	"version":"0.1-alpha", // this much match the version in index.js
	"moduleList":[ // modules are listed in an array
		"github.com/wgytcraft/errors.git" // error handler
	],
	"errorHandler":"wgytcraft/errors"
}
```
github.com/wgytcraft/errors.git would be put in the wgytcraft/errors folder which is what the errorHandler variable and the pages object will refer to. I think we should also add a shorthand version for all repos published by wgytcraft.
Example:
```json
{
	"version":"0.1-alpha", // this much match the version in index.js
	"moduleList":[ // modules are listed in an array
		"errors" // error handler
	],
	"errorHandler":"errors"
}
```

and then we would import the modules.
## import the modules
how would we do this?
we should probably use import('') functions for this
@grahamsh-llk could help (sorry for the mention)

**Comment by wgyt** [comment]
Here is the cool awesome code by graham:
```javascript
var Git = require("nodegit");
modules = JSON.parse(fs.readFileSync('modules/modules.json'))
for (module of modules.moduleList) {
Git.Clone(`https://github.com/wgytcraft/${module}`, `./modules/${module}`).then(function(repository) {
  // Work with the repository object here.
});
}
```

**Comment by wgyt** [comment]
@grahamsh-llk the code works, but it throws an error when stopping and restarting the server if the folder still exists
```
(node:1232) UnhandledPromiseRejectionWarning: Error: './modules/wgytcraft/errors' exists and is not an empty directory
(node:1232) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1232) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Should we just empty the folders in the modules folder?

**Comment by wgyt** [comment]
will be resolved in #4 with help from @grahamsh-llk and @thecolaber
